ckanext/cfpb_extrafields/plugin.py

```diff
@@ -76,9 +76,16 @@ def _redirect_to_edit_on_change(self, resource, field):
     def _delete_and_rebuild_datadict(self, resource):
         import json
         if 'datadict' in resource and 'id' in resource:
-            record = resource['datadict']
-            resource.pop('datadict')
-            json_record = json.loads(record)
+            record = resource.pop('datadict')
+
+            try:
+                json_record = json.loads(record)
+            except ValueError as e:
+                # Invalid JSON, so don't remove old data
+                error_message = "Error saving data dictionary: {0}.  Data was: {1}".format(e, record)
+                flash_error(error_message)
```

Comment:
Awesome, but we probably want more explicit instructions, so that people know to avoid special characters and limit the amount of text?